src/internal/observable/of.ts

```diff
@@ -19,6 +19,7 @@ export function of<T, T2, T3, T4, T5, T6, T7, T8>(a: T, b: T2, c: T3, d: T4, e:
 export function of<T, T2, T3, T4, T5, T6, T7, T8, T9>(a: T, b: T2, c: T3, d: T4, e: T5, f: T6, g: T7, h: T8, i: T9, scheduler?: SchedulerLike):
   Observable<T | T2 | T3 | T4 | T5 | T6 | T7 | T8 | T9>;
 export function of<T>(...args: Array<T | SchedulerLike>): Observable<T>;
+export function of<T>(...args: Array<unknown | SchedulerLike>): Observable<T>;
```

Comment:
We would need a dtslint test to cover this.

But this *looks* like you're trying to get the user to force a generic in the signature like so:  `of<number>(1,2,3,4,5,6,7,'whatever',9,10,11)`, that's an antipattern, really,  I think we should just return `Observable<{}>` or the like, and force the user to cast themselves. `Observable<unknown>` might be better.

Reply:
Sorry @benlesh , I don't quite follow. 
what I try to solve here is to fix [this test](https://github.com/ReactiveX/rxjs/pull/4445/files#diff-d0ec8bfb3126e86588f3e57115b9a6feR56) that I commented out because it does not behave as it should be